apps/aevm/lib/aevm_util.ex

```diff
@@ -0,0 +1,374 @@
+defmodule AevmUtil do
+  @moduledoc """
+    Module containing all the VM utility functions
+  """
+
+  use Bitwise
+
+  require AevmConst
+  require OpCodes
+  require GasCodes
+
+  @spec default_opts :: map()
+  def default_opts do
+    %{
+      :execute_calls => true
+    }
+  end
+
+  @spec stop_exec(map()) :: map()
+  def stop_exec(state) do
+    code = State.code(state)
+    State.set_pc(byte_size(code), state)
+  end
+
+  def sdiv(_value1, 0), do: 0
+  def sdiv(0, -1), do: AevmConst.neg2to255()
+
+  @spec sdiv(integer(), integer()) :: non_neg_integer()
+  def sdiv(value1, value2) do
+    <<svalue1::integer-signed-256>> = <<value1::integer-unsigned-256>>
+    <<svalue2::integer-signed-256>> = <<value2::integer-unsigned-256>>
+    div(svalue1, svalue2) &&& AevmConst.mask256()
+  end
+
+  @spec smod(integer(), integer()) :: non_neg_integer()
+  def smod(value1, value2) do
+    <<svalue1::integer-signed-256>> = <<value1::integer-unsigned-256>>
+    <<svalue2::integer-signed-256>> = <<value2::integer-unsigned-256>>
+    result = rem(rem(svalue1, svalue2 + svalue2), svalue2)
+    result &&& AevmConst.mask256()
+  end
+
+  def pow(op1, op2) when is_integer(op1) and is_integer(op2) and op2 >= 0, do: pow(1, op1, op2)
```

Comment:
You could make a specific Module for such long guards and combine them into one. Check Defining custom guard expressions [here](https://hexdocs.pm/elixir/master/guards.html).

Reply:
I agree a `non_neg_integer` guard would be nice.